ts/private/ts_project.bzl

```diff
@@ -85,19 +85,23 @@ See https://github.com/aspect-build/rules_ts/issues/228 for more details.
     # Add user specified arguments *before* rule supplied arguments
     arguments.add_all(ctx.attr.args)
 
-    outdir = _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package
-    if outdir == "":
-        outdir = "."
+    outdir = _lib.join(
+        ctx.label.workspace_root,
+        _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package,
+    )
+    tsconfig_path = ctx.file.tsconfig.path.removeprefix("%s/" % ctx.bin_dir.path)
     arguments.add_all([
         "--project",
-        ctx.file.tsconfig.short_path,
+        tsconfig_path,
         "--outDir",
         outdir,
         "--rootDir",
         _lib.calculate_root_dir(ctx),
     ])
     if len(typings_outs) > 0:
         declaration_dir = _lib.join(ctx.label.package, typings_out_dir)
+        if ctx.label.workspace_root:
```

Comment:
Can we remove the `if`s similar to `outdir`?

Reply:
Done.